File: src/test/java/br/com/blz/testjava/entities/InventoryTest.java
package br.com.blz.testjava.entities;

import static org.junit.jupiter.api.Assertions.assertEquals;

import java.util.ArrayList;
import java.util.List;

import org.junit.jupiter.api.Test;

class InventoryTest {

	@Test
	void getQuantityTest() {
		var inventory = new Inventory();
		inventory.setWarehouses(createWarehouses());
		assertEquals(5, inventory.getQuantity());
	}
	
	private List<Warehouse> createWarehouses(){
		var warehouses = new ArrayList<Warehouse>();
		var warehouse1 = new Warehouse();
		warehouse1.setQuantity(2);
		var warehouse2 = new Warehouse();
		warehouse2.setQuantity(3);
		warehouses.add(warehouse1);
		warehouses.add(warehouse2);
		return warehouses;
	}
}

File: src/test/kotlin/br/com/blz/testjava/domain/entities/InventoryTest.kt
package br.com.blz.testjava.domain.entities

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test

internal class InventoryTest {

  @Test
  fun `An inventory with no quantity on warehouses must sum quantity of 0`() {
    val warehouses = mutableListOf<Warehouse>(
      Warehouse("PR", 0, WarehouseType.ECOMMERCE),
      Warehouse("SP", 0, WarehouseType.PHYSICAL_STORE)
    )
    val inventory = Inventory(warehouses)

    assertEquals(0, inventory.getQuantity())
  }

  @Test
  fun `An inventory with 5 quantity in all warehouses must sum quantity of 5`() {
    val warehouses = mutableListOf<Warehouse>(
      Warehouse("PR", 2, WarehouseType.ECOMMERCE),
      Warehouse("SP", 3, WarehouseType.PHYSICAL_STORE)
    )
    val inventory = Inventory(warehouses)

    assertEquals(5, inventory.getQuantity())
  }

  @Test
  fun `An inventory with 5 quantity in only one warehouse must sum quantity of 5`() {
    val warehouses = mutableListOf<Warehouse>(
      Warehouse("PR", 5, WarehouseType.ECOMMERCE),
      Warehouse("SP", 0, WarehouseType.PHYSICAL_STORE)
    )
    val inventory = Inventory(warehouses)

    assertEquals(5, inventory.getQuantity())
  }

}
